File: API/lb/api/connectors/LetterAttachmentClient.php
<?php


class LetterAttachmentClient
{
	public $application;

	
	public function uploadAttachment($letterId, $attachment) {
				
		$jsonParameters = array();
		$jsonParameters["letterId"] = $letterId;
		$jsonParameters["index"] = $attachment->getIndex();
		$jsonParameters["X-Progress-ID"] = $letterId . "_" . $attachment->getIndex();
		$jsonParameters["filename"] = $attachment->getFilename();

		$session = $this->application->session;
    	$jsonParameters["sessionId"] = $session->sessionId;
    	$jsonParameters["userId"] = $session->userId;
    	
		$uploadUrl = APIResourcesManager::$uploadUrl 
			. "?sessionId=" . $session->sessionId
			. "&userId=" . $session->userId
			. "&letterId=" . $jsonParameters["letterId"]
			. "&index=" . $jsonParameters["index"]
			. "&X-Progress-ID=" . $jsonParameters["X-Progress-ID"]
			. "&filename=" . $jsonParameters["filename"]
			;
    	
		$session->executePostByUrlWithFile($uploadUrl, $jsonParameters, $attachment->getPathFilename());
		
//		if (attachment.isSigned()) {
//			session.createApplicationClient()
//				.addAttachmentSignatureRemote(letterId, attachment.getIndex(), attachment.getBase64Signature());
//		}
		
	}
		
}
?>

File: API/lb_clientAPI_sources/lb/api/connectors/LetterAttachmentClient.java
package lb.api.connectors;

import java.io.IOException;

import lb.api.APIResourcesManager;
import lb.api.beans.DraftAttachment;
import lb.api.exceptions.RemoteException;

import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.json.JSONException;
import org.json.JSONObject;

public class LetterAttachmentClient {
	
	@SuppressWarnings("unused")
	private static Log log = LogFactory.getLog(LetterAttachmentClient.class);
	
	public static String uploadUrl = APIResourcesManager.getFileUploadRestfulURL();
	
	protected SessionClient session;

	public LetterAttachmentClient(SessionClient session) {
		this.session = session;
	}
	
	public void uploadAttachment(String letterId, DraftAttachment attachment) throws IOException, 
	JSONException, RemoteException {
		
		JSONObject jsonParameters = new JSONObject();
		jsonParameters.put("letterId", letterId);
		jsonParameters.put("index", "" + attachment.getIndex());
		jsonParameters.put("X-Progress-ID", letterId + "_" + attachment.getIndex());
		jsonParameters.put("filename", attachment.getFilename());

		session.executeApplicationPost(uploadUrl, jsonParameters, attachment.getHttpBody());
		
		if (attachment.isSigned()) {
			session.createApplicationClient()
				.addAttachmentSignatureRemote(letterId, attachment.getIndex(), attachment.getBase64Signature());
		}
		
	}
	
}
